include/slang/text/CharInfo.h

```diff
@@ -194,6 +194,16 @@ constexpr const char* utf8Decode(const char* b, uint32_t* c, int* e, int& comput
     *e |= (uc(b[3])) >> 6;
     *e ^= 0x2a; // top two bits of each tail byte correct?
     *e >>= shifte[len];
+    // For normal path, this should not be checked
```

Comment:
The comment above describes this function as branchless; I think we should keep it that way. You can put this handling in Lexer::scanUTF8Char instead.

Reply:
Added.